scripts/autotools/libbson/FindDependencies.m4

```diff
@@ -1,3 +1,12 @@
+if test "$os_win32" != "yes"; then
+    PHP_MONGODB_BUNDLED_CFLAGS="$PHP_MONGODB_BUNDLED_CFLAGS -D_DEFAULT_SOURCE"
+fi
+
+# Enable macOS extensions for strlcpy and arc4random
+if test "$os_darwin" = "yes"; then
+    PHP_MONGODB_BUNDLED_CFLAGS="$PHP_MONGODB_BUNDLED_CFLAGS -D_DARWIN_C_SOURCE=1"
+fi
```

Comment:
@eramongodb: Ah, I didn't realize you had removed this from https://github.com/mongodb/mongo-c-driver/pull/920 before merging. We actually don't test macOS in CI at the moment, so I can't say for certain if this was required. I'm happy to remove it for now and we can revisit if it comes up down the line.